app/leaderboard/[duration]/Leaderboard.tsx

```diff
@@ -51,19 +51,31 @@ export default function Leaderboard(props: Props) {
   const search = searchParams.get("search") || "";
 
   const roles = searchParams.get("roles")?.split(",") || [];
+
   const ordering = ORDERING_OPTIONS.find(
     (option) => option.value === searchParams.get("ordering"),
   ) || { value: "points", text: "Points" };
+
   const isReversed = searchParams.get("isReversed") === "true";
 
   let resultSet = props.data;
 
-  if (roles.length) {
-    resultSet = resultSet.filter((a) => roles.includes(a.user.role));
+  if (ordering.value) {
+    resultSet = props.data.sort((a, b) => {
+      const aValue = a.highlights[ordering.value as keyof typeof a.highlights];
+      const bValue = b.highlights[ordering.value as keyof typeof a.highlights];
+
+      const aNumericValue = typeof aValue === "number" ? aValue : 0;
+      const bNumericValue = typeof bValue === "number" ? bValue : 0;
+
+      return isReversed
+        ? aNumericValue - bNumericValue
+        : bNumericValue - aNumericValue;
```

Comment:
The reverse would not work if no ordering is specified since it's part of the if block

Reply:
![image](https://github.com/user-attachments/assets/979ba1bf-04e9-4a75-8153-19716239fccc)

can u check now @rithviknishad

Reply:
we could simply return `aNumericValue - bNumericValue` instead.

And in another if block we could do `resultSet = resultSet.toReversed()` if `isReversed` just like before. would make the code more readable.

Reply:
I did because If we do like that  the iteration will happen 2 times. in our case it only happens once. if you prefer readable code let me know i will change it